akka-projection-jdbc/src/test/scala/akka/projection/jdbc/JdbcProjectionSpec.scala

```diff
@@ -895,50 +897,64 @@ class JdbcProjectionSpec
         offsetOpt shouldBe empty
       }
 
-      val failingProjection =
+      def buildProjection(failPredicate: Long => Boolean = _ => false) =
         JdbcProjection
           .atLeastOnce(
             projectionId,
             sourceProvider = sourceProvider(system, entityId),
             jdbcSessionFactory,
-            handler = () => new ConcatHandler(_ == 4))
+            handler = () => new ConcatHandler(failPredicate))
           .withSaveOffset(2, 1.minute)
 
+      val failingProjection = buildProjection(_ == 4)
+
       withClue("check - offset is empty") {
         val offsetOpt = offsetStore.readOffset[Long](projectionId).futureValue
         offsetOpt shouldBe empty
       }
 
+      withClue("check: projection handles 3 elements") {
+        projectionTestKit.runWithTestSink(failingProjection) { sinkProbe =>
+          sinkProbe.request(3)
+          eventually {
+            val concatStr = findById(entityId)
+            concatStr.text shouldBe "abc|def|ghi"
+          }
+        }
+      }
```

Comment:
This step is new: demand only 3 elements and await until all three have been handled.